### ISSUE TITLE
Instant yields applied multiple times when a city kills a unit

### ISSUE THREAD

**Issue by JohannesGaessler** [LoneGazebo/Community-Patch-DLL#6019]
_1. Mod version (i.e Date - 4/23):_
12/20

_2. Mod list (if using Vox Populi only, leave blank):_
* Wonder Race
* Faster Aircraft Animations
* I modified some parameters of Vox Populi:
  * Warmonger Score decay: -5 -> -15.
   * Pantheons non-exclusive.
   * Pathfinders get goody hut picker promotion for free.
* I'm playing on Linux via WINE (should not make a difference for the mod's logic).

_3. Error description:_ 
When a ~~puppeted~~ City kills an enemy Unit Instant Yields (like the Science from the Dominance Policy) are applied ~~too often~~ multiple times.
~~I do not know if non-puppeted Cities are affected as well.~~
Non-puppeted Cities can be affected as well.

_4. Steps to reproduce (optional):_
~~Kill an enemy Unit with a puppeted City.~~
After conquering a City, use it to kill a ~~naval~~ Unit of the original City owner.

[debug.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/3998602/debug.zip)

**Comment by bigcat88** [comment]
What you mean by `too often`? Each time you kill a unit it must apply.

**Comment by JohannesGaessler** [comment]
What I meant is that the Instant Yields get applied multiple times instead of just once.

**Comment by JohannesGaessler** [comment]
In another game I've gotten more insight to the bug described in the OP.
I think the bug is actually unrelated to puppeted Cities.
Instead it seems to appear when you use a conquered City to kill a Unit of the original City owner.
I've attached another zip file.
[debug-2.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/4006871/debug-2.zip)

**Comment by JohannesGaessler** [comment]
I also got the bug described in the OP when I killed a Barbarian naval Unit with a City:
[debug-3.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/4007117/debug-3.zip)

**Comment by ilteroi** [comment]
cannot reproduce (with culture from authority opener at least)

**Comment by LoneGazebo** [comment]
Are you sure you're not just seeing Kill and Global Kill instant yields? They're each applied.

**Comment by JohannesGaessler** [comment]
Yes, I am sure that I am not confused about Kill vs Global Kill yields.
When I get the bug I get the Kill yields (like from the Dominance Policy) in essentially each of my cities.
I am very sure that this is not intended.

I do not know the exact conditions for reproducing the bug.
I get the bug almost every time when I kill a Unit with a City that I did not found.
In some rare cases I also get it when I kill a Unit with a City that I did found or with one of my Units.
